--- packages.orig/javascriptcore.vapi	2011-02-10 06:49:30.000000000 +0000
+++ packages/javascriptcore.vapi	2012-11-10 08:06:30.000000000 +0000
@@ -23,6 +23,8 @@
  * http://live.gnome.org/Seed for more information.
  */
 
+// Retrieved from: http://gitorious.org/seed-vapi/seed-vapi/blobs/master/javascriptcore.vapi
+
 [CCode (lower_case_cprefix = "js_", cheader_filename = "JavaScriptCore/JavaScript.h")]
 namespace JSCore {
 	[Compact]
@@ -37,7 +39,7 @@
 
 
 	[Compact]
-	[CCode (cname = "void")]
+	[CCode (cname = "const struct OpaqueJSContext")]
 	public class Context {
 		/* Script Evaluation */
 		[CCode (cname = "JSEvaluateScript")]
@@ -161,7 +163,7 @@
 	                                                                       JSCore.Object function,
 	                                                                       JSCore.Object thisObject,
                                                                            [CCode (array_length_pos=3.9, array_length_type="size_t")]
-	                                                                       JSCore.Value[] arguments,
+	                                                                       JSCore.ConstValue[] arguments,
 	                                                                       out JSCore.Value exception);
 
 	[CCode (has_target = false)]
@@ -275,7 +277,7 @@
 
 	[Compact]
 	/* FIXME: free_function causes a warning */
-	[CCode (cname = "void", free_function = "")]
+	[CCode (cname = "const struct OpaqueJSValue", free_function = "")]
 	public class Value {
 		[CCode (cname = "JSValueMakeUndefined")]
 		public Value.undefined (Context ctx);
@@ -330,7 +332,7 @@
 		public bool to_boolean (Context ctx);
 
 		[CCode (cname = "JSValueToNumber", instance_pos=1.1)]
-		public bool to_number (Context ctx, Value *exception);
+		public double to_number (Context ctx, Value *exception);
 
 		[CCode (cname = "JSValueToStringCopy", instance_pos=1.1)]
 		public JSCore.String to_string_copy (Context ctx, Value *exception);
@@ -345,10 +347,15 @@
 		public void unprotect (Context ctx);
 	}
 
+	[Compact]
+	[CCode (cname = "ConstOpaqueJSValue", free_function = "")]
+	public class ConstValue : JSCore.Value
+	{
+	}
 
 	[Compact]
 	/* FIXME: causes a warning */
-	[CCode (cname = "void", free_function = "")]
+	[CCode (cname = "struct OpaqueJSValue", free_function = "")]
 	public class Object: JSCore.Value {
 		[CCode (cname = "JSObjectMake")]
 		public Object (Context ctx, Class js_class, void *data);